src/Http/Controllers/PagesApiController.php

```diff
@@ -27,26 +28,45 @@ public function list(PageListingRequest $request): JsonResponse
         return response()->json($pages);
     }
 
-    public function read(PageViewRequest $request): JsonResponse
+    public function create(PageCreateRequest $request): JsonResponse
     {
-//        $page = $this->pageService->getBySlug($request->getSlug());
-//        return response()->json($page);
-        return $this->sendError('Not implemented', 404);
+        $slug = $request->getParamSlug();
+        $title = $request->getParamTitle();
+        $content = $request->getParamContent();
+
+        $user = Auth::user();
+        $page = $this->pageService->insert($slug,$title,$content,$user->id);
+        return response()->json($page);
     }
 
-    public function insert(PageInsertRequest $request): JsonResponse
+    public function update(PageUpdateRequest $request): JsonResponse
     {
         return $this->sendError('Not implemented', 404);
     }
 
     public function delete(PageDeleteRequest $request): JsonResponse
     {
-        return $this->sendError('Not implemented', 404);
+        $slug = $request->getParamSlug();
+
+        $deleted = $this->pageService->delete($slug);
```

Comment:
You should name this function `deleteBySlug`, or maybe better call first `findBySlug` and then `delete` that found page?

Reply:
Renamed it to deleteBySlug